semgrep-core/src/experiments/synthesizing/Synthesizer.mli

```diff
@@ -5,12 +5,14 @@ val synthesize_patterns :
 
 (* "start row:start col-end row:end col"+ file ((range list)@[file]) -> pattern list *)
 (* Generates a pattern from several target ranges that matches each target.
-  Returns the detected language, the list of targest, and the Pattern list.
+   Returns the detected language, the list of targest, and the Pattern list.
 *)
 
 val generate_pattern_from_targets :
   Config_semgrep.t -> string list -> Lang.t * AST_generic.any list * Pattern.t
 
+val synthesize_from_diff_list : string -> unit
```

Comment:
You should document what's not obvious about the function. Here this would be the format or origin of the string argument, and also ideally explain what it's for.